src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java

```diff
@@ -0,0 +1,85 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.PBAValidationService;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.YesOrNo.NO;
+
+@ExtendWith(MockitoExtension.class)
+class CaseSubmissionPbaValidateMidEventHandlerTest {
+
+    @Mock
+    private PBAValidationService pbaValidationService;
+
+    @InjectMocks
+    private CaseSubmissionPbaValidateMidEventHandler handler;
+
+    @Test
+    void testCanHandle() {
+        assertTrue(handler.canHandle(CallbackType.MID_EVENT, CaseType.CONSENTED, EventType.APPLICATION_PAYMENT_SUBMISSION));
+        assertTrue(handler.canHandle(CallbackType.MID_EVENT, CaseType.CONTESTED, EventType.APPLICATION_PAYMENT_SUBMISSION));
+        assertFalse(handler.canHandle(CallbackType.ABOUT_TO_START, CaseType.CONTESTED, EventType.APPLICATION_PAYMENT_SUBMISSION));
+    }
+
+    @Test
+    void testHandle_ValidPbaNumber_thenHandlerCanHandle() {
+        FinremCaseDetails caseDetails = new FinremCaseDetails();
+        FinremCaseData caseData = new FinremCaseData();
+        caseData.setHelpWithFeesQuestion(NO);
+        caseData.setPbaNumber("ValidPbaNumber");
+        caseDetails.setData(caseData);
+
+        Mockito.when(pbaValidationService.isValidPBA(any(), eq("ValidPbaNumber"))).thenReturn(true);
+
+        FinremCallbackRequest request = FinremCallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .build();
+        var response = handler.handle(request, "userAuthorisation");
+
+        assertThat(response.getErrors()).isEmpty();
+        assertEquals(caseData, response.getData());
+    }
+
+    @Test
+    void testHandle_InvalidPbaNumber_thenReturnError() {
+        FinremCaseDetails caseDetails = new FinremCaseDetails();
+        FinremCaseData caseData = new FinremCaseData();
+        caseData.setHelpWithFeesQuestion(NO);
+        caseData.setPbaNumber("InvalidPbaNumber");
+        caseDetails.setData(caseData);
+
+        Mockito.when(pbaValidationService.isValidPBA(any(), eq("InvalidPbaNumber"))).thenReturn(false);
+
+        FinremCallbackRequest request = FinremCallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .build();
+        var response = handler.handle(request, "userAuthorisation");
+
+        System.out.println("Response: " + response.getErrors());
+        System.out.println("Response: " + response);
+
+
+        assertThat(response).isNotNull();
+        assertThat(response.getErrors()).isNotEmpty();
```

Comment:
I would make the test more precise. There should only be 1 error message:
`.size(), is(1));`
Similar thing with the following line. Use
`assertThat(response.getErrors.get(0)).isEqualTo(....`

Reply:
[96b834e](https://github.com/hmcts/finrem-case-orchestration-service/pull/1610/commits/96b834ed0ba13abb53a65ceb5bb5610485f66cc0)